Lib/multiprocessing/shared_memory.py

```diff
@@ -1,225 +1,316 @@
-"Provides shared memory for direct access across processes."
+"""Provides shared memory for direct access across processes.
+
+The API of this package is currently provisional. Refer to the
+documentation for details.
+"""
 
 
 __all__ = [ 'SharedMemory', 'PosixSharedMemory', 'WindowsNamedSharedMemory',
-            'ShareableList', 'shareable_wrap',
-            'SharedMemoryServer', 'SharedMemoryManager', 'SharedMemoryTracker' ]
+            'ShareableList',
+            'SharedMemoryServer', 'SharedMemoryManager' ]
 
 
-from functools import reduce
+from functools import partial
 import mmap
-from .managers import DictProxy, SyncManager, Server
+from .managers import dispatch, BaseManager, Server, State, ProcessError
 from . import util
 import os
-import random
 import struct
-import sys
+import secrets
 try:
-    from _posixshmem import _PosixSharedMemory, Error, ExistentialError, O_CREX
+    import  _posixshmem
+    from os import O_CREAT, O_EXCL, O_TRUNC
 except ImportError as ie:
-    if os.name != "nt":
-        # On Windows, posixshmem is not required to be available.
-        raise ie
-    else:
-        _PosixSharedMemory = object
-        class ExistentialError(BaseException): pass
-        class Error(BaseException): pass
-        O_CREX = -1
+    O_CREAT, O_EXCL, O_TRUNC = 64, 128, 512
+
+O_CREX = O_CREAT | O_EXCL
+
+if os.name == "nt":
+    import ctypes
+    from ctypes import wintypes
+
+    kernel32 = ctypes.windll.kernel32
+
+    class MEMORY_BASIC_INFORMATION(ctypes.Structure):
+        _fields_ = (
+            ('BaseAddress',       ctypes.c_void_p),
+            ('AllocationBase',    ctypes.c_void_p),
+            ('AllocationProtect', wintypes.DWORD),
+            ('RegionSize',        ctypes.c_size_t),
+            ('State',             wintypes.DWORD),
+            ('Protect',           wintypes.DWORD),
+            ('Type',              wintypes.DWORD)
+        )
+
+    PMEMORY_BASIC_INFORMATION = ctypes.POINTER(MEMORY_BASIC_INFORMATION)
+    PAGE_READONLY = 0x02
+    PAGE_EXECUTE_READWRITE = 0x04
+    INVALID_HANDLE_VALUE = -1
+    FILE_ALREADY_EXISTS = 183
```

Comment:
This constant shouldn't be necessary once other mistakes are corrected. But it should be named `ERROR_ALREADY_EXISTS` if it stays. It's confusing to readers if we invent our own names for well-known error codes.

Reply:
Thanks for catching this detail about the appropriate name.  Changed in commit 868b83d681c3e11dd6de22631ac61493a5b73ded (at least until it is no longer needed).

Reply:
These have now been (re-)moved to _winapi per your later suggestion as part of commit 68785333a25d902a42252ace3a8268357efd6094.